src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -85,7 +86,8 @@ case class AppDefinition(
   def portIndicesAreValid(): Boolean = {
     val validPortIndices = 0 until hostPorts.size
     healthChecks.forall { hc =>
-      validPortIndices contains hc.portIndex
+      (hc.protocol == Protocol.COMMAND && hc.portIndex == 0) ||
```

Comment:
So portIndex will also have no influence on the PORT0... environment variables, correct? Then I think you are right. Then we can drop the check. I somehow assumed PORT0 is the portIndex port.